hass_nabucasa/cloud_api.py

```diff
@@ -98,6 +98,8 @@ async def log_response(
 @_log_response
 async def async_create_cloudhook(cloud: Cloud[_ClientT]) -> ClientResponse:
     """Create a cloudhook."""
+    if TYPE_CHECKING:
+        assert cloud.id_token is not None
     return await cloud.websession.post(
         f"https://{cloud.cloudhook_server}/generate",
         headers={AUTHORIZATION: cloud.id_token, USER_AGENT: cloud.client.client_name},
```

Comment:
Wouldn't something like this be shorter?
```suggestion
        headers={AUTHORIZATION: str(cloud.id_token or ''), USER_AGENT: cloud.client.client_name},
```
This could be also extracted to a function like `getOrEmpty(cloud.id_token)`. Or, maybe even `getHearders(cloud)` as it seems the headers are the same everywhere.

Reply:
I'd look into improving the type information, since it looks like there's a decorator that is already validating the token. I don't think any runtime code needs to change.

Reply:
> Wouldn't something like this be shorter?

Yes, this is shorter, but as we know it is a string, making it a string again is not needed/wanted.

Reply:
Could have still been a `getHeaders` function, so you don't have to repeat this so many times 🙂